### DISCUSSION TITLE
Zap Restful api with swagger

### DISCUSSION THREAD

**Discussion by rrogovski** [zigzap/zap#147] [General]
Has anyone used ZAP for this purpose, restful api with swagger documentation?

**Comment by Blankeos** [comment]
I asked it before, but I don't think it's one of its goals: https://github.com/zigzap/zap/issues/116 (It is just an HTTP Server afterall).

I don't know if it's possible to create a router wrapper around Zap and call that a backend framework.

I also wonder why it's not more sought-after by most backend frameworks though. The API still has to be consumed by JavaScript at some point and having OpenAPI makes it possible to consume it with typesafety ([Orval](https://orval.dev/), [OpenAPI TS](https://openapi-ts.dev/)).

It's kinda why I stick to the JS ecosystem for backends that require this since it's there out-of-the-box. (e.g. [Elysia](https://elysiajs.com/), [Hono](https://github.com/rhinobase/hono-openapi)).

If you really want to use a compiled language. Maybe try Go (https://github.com/swaggest/openapi-go) or Rust (https://www.rspc.dev/) - I think most pleasant tbh.
